From d33642daf10bd2ebe512c25eb686fe00396849bc Mon Sep 17 00:00:00 2001
From: Michael Bestas <mkbestas@lineageos.org>
Date: Thu, 25 Jan 2018 23:09:55 +0200
Subject: [PATCH 09/13] legacy: Import hci_qcomm_init policies

* Based on our old policy
* Trimmed down, init.qcom.bt.sh must be removed device side
  and hci_qcomm_init should run directly as a service

Change-Id: Iab96e737f9577056a4196090b8513f0ca9efe0f3
---
 legacy-common/bluetooth_loader.te | 23 +++++++++++++++++++++++
 legacy-common/file_contexts       |  3 +++
 2 files changed, 26 insertions(+)
 create mode 100644 legacy-common/bluetooth_loader.te

diff --git a/legacy-common/bluetooth_loader.te b/legacy-common/bluetooth_loader.te
new file mode 100644
index 0000000..b7ccd98
--- /dev/null
+++ b/legacy-common/bluetooth_loader.te
@@ -0,0 +1,23 @@
+type bluetooth_loader, domain;
+type bluetooth_loader_exec, exec_type, file_type;
+
+# Started by init
+init_daemon_domain(bluetooth_loader)
+
+# Set persist.service.bdroid.* and bluetooth.* property values
+get_prop(bluetooth_loader, bluetooth_prop)
+
+# Access the serial device
+allow bluetooth_loader serial_device:chr_file rw_file_perms;
+
+# And the smd device
+allow bluetooth_loader smd_device:chr_file rw_file_perms;
+
+# And qmuxd
+allow bluetooth_loader qmuxd_socket:dir create_dir_perms;
+allow bluetooth_loader qmuxd_socket:sock_file create_file_perms;
+allow bluetooth_loader qmuxd:unix_stream_socket connectto;
+
+userdebug_or_eng(`
+  diag_use(bluetooth_loader)
+')
diff --git a/legacy-common/file_contexts b/legacy-common/file_contexts
index 6e692d9..c57d57e 100644
--- a/legacy-common/file_contexts
+++ b/legacy-common/file_contexts
@@ -62,3 +62,6 @@
 
 # Old camera sockets
 /data/cam_socket[12]                                                          u:object_r:camera_socket:s0
+
+# Bluetooth
+/(vendor|system/vendor)/bin/hci_qcomm_init                                    u:object_r:bluetooth_loader_exec:s0
-- 
2.7.4

